hack/test-go.sh

```diff
@@ -288,6 +288,9 @@ elif [[ -n "${coverage_output_dir}" ]]; then
     # we can clean up all of the coverage reports at once as they all exist in subdirectories of ${coverage_output_dir}/${OS_GO_PACKAGE}
     # and they are the only files found in those subdirectories
     rm -rf "${coverage_output_dir:?}/${OS_GO_PACKAGE}"
+
+elif [[ -n "${DEBUG-}" ]]; then
+    dlv test ${test_packages}
```

Comment:
Do we also want to have a binary check for `dlv` and spit out our go-to 'binary is missing' message here if we don't find it?